serde_derive/src/internals/case.rs

```diff
@@ -6,6 +6,9 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+//! Utility code to assist converting the case of the source (e.g. `my-field`, `MY_FIELD`) to the
+//! rust-styled field/variant (e.g. `my_field`, `MyType`).
```

Comment:
This code actually only converts in the opposite direction, from name in Rust to name in the input, so I find this explanation misleading even if technically correct. The conversion from name in the input to name in Rust happens elsewhere by emitting a match-expression.

Reply:
Can you check you're happy with this now.